### DISCUSSION TITLE
replicate a whole OSX crash with wezterm

### DISCUSSION THREAD

**Discussion by yeled** [wezterm/wezterm#3889] [General]
Hi,

I started using Wezterm this week but it _crashes_ my whole OS. It really bogs down when I sometimes click a URI in https://github.com/weechat/weechat and then eventually interaction with mouse or keyboard becomes sluggish and freezes. Then it reboots.

The `WEZTERM_LOG=debug` output is really not enlightening. It has a lot of private information so I won't post it here.

My config is here https://gist.github.com/yeled/92fdbecc7249c92ddf337a059b776178 and I'm using

Here is the panic from the OS

```
panic(cpu 14 caller 0xffffff800c579941): userspace watchdog timeout: no successful checkins from WindowServer in 120 seconds
service: logd, total successful checkins in 11389 seconds: 1110, last successful checkin: 60 seconds ago
service: WindowServer, total successful checkins in 11350 seconds: 1114, last successful checkin: 120 seconds ago
service: remoted, total successful checkins in 11389 seconds: 1137, last successful checkin: 0 seconds ago
service: opendirectoryd, total successful checkins in 11389 seconds: 1139, last successful checkin: 0 seconds ago
service: configd, total successful checkins in 11389 seconds: 1138, last successful checkin: 0 seconds ago

Panicked task 0xffffff9bbbccc478: 4 threads: pid 120: watchdogd
Backtrace (CPU 14), panicked thread: 0xffffff96f3f36598, Frame : Return Address
0xfffffff4716ff5a0 : 0xffffff800926f4fd 
0xfffffff4716ff5f0 : 0xffffff80093c3c54 
0xfffffff4716ff630 : 0xffffff80093b36d9 
0xfffffff4716ff690 : 0xffffff800920f951 
0xfffffff4716ff6b0 : 0xffffff800926f7dd 
0xfffffff4716ff7a0 : 0xffffff800926ee87 
0xfffffff4716ff800 : 0xffffff80099dcf73 
0xfffffff4716ff8f0 : 0xffffff800c579941 
0xfffffff4716ff900 : 0xffffff800c5795dc 
0xfffffff4716ff920 : 0xffffff800c5786b9 
0xfffffff4716ffa50 : 0xffffff8009954cee 
0xfffffff4716ffbb0 : 0xffffff80093701e9 
0xfffffff4716ffcc0 : 0xffffff800924975a 
0xfffffff4716ffd60 : 0xffffff80092604a2 
0xfffffff4716ffdd0 : 0xffffff8009260b27 
0xfffffff4716ffef0 : 0xffffff8009398f53 
0xfffffff4716fffa0 : 0xffffff800920fdb6 
      Kernel Extensions in backtrace:
         com.apple.driver.watchdog(1.0)[BD08CE2D-77F5-358C-8F0D-A570540A0BE7]@0xffffff800c577000->0xffffff800c579fff

Process name corresponding to current thread (0xffffff96f3f36598): watchdogd

Mac OS version:
22F66

[ ... ]
```

There are some choice errors I can pick out of the logfile, such as

```
22:43:29.819  DEBUG  config > Reloaded configuration! generation=1
22:43:29.972  DEBUG  config > Reloaded configuration! generation=2
22:43:29.981  DEBUG  window::os::macos::app > application_will_finish_launching
22:43:30.014  DEBUG  wezterm_gui::frontend  > workspace is default, fixup windows
22:43:30.015  DEBUG  window::os::macos::app > application_did_finish_launching
22:43:30.026  DEBUG  wezterm_font::ftwrap   > set_char_size computing 12 dpi=72 (pixel height=12)
22:43:30.039  DEBUG  window::os::macos::connection > NSAppearanceName is NSAppearanceNameDarkAqua
22:43:30.048  DEBUG  wezterm_gui::termwindow       > Resized { dimensions: Dimensions { pixel_width: 984, pixel_height: 832, dpi: 144 }, window_state: (empty), live_resizing: false }
22:43:30.053  DEBUG  wezterm_font::ftwrap          > set_char_size computing 12 dpi=144 (pixel height=24)
22:43:30.055  DEBUG  wezterm_term::screen          > resize screen to 80x24 dpi=144
22:43:30.055  DEBUG  wezterm_term::screen          > resize screen to 80x24 dpi=144
22:43:30.069  DEBUG  wezterm_gui::termwindow       > Resized { dimensions: Dimensions { pixel_width: 984, pixel_height: 780, dpi: 144 }, window_state: (empty), live_resizing: false }
22:43:30.074  DEBUG  wezterm_gui::termwindow       > NeedRepaint
22:43:30.076  DEBUG  window::os::macos::window     > EGL init failed: with_egl_lib failed: /Applications/WezTerm.app/Contents/MacOS/libEGL.dylib: dlopen(/Applications/WezTerm.app/Contents/MacOS/libEGL.dylib, 0x0002): tried: '/Applications/WezTerm.app/Contents/MacOS/libEGL.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Applications/WezTerm.app/Contents/MacOS/libEGL.dylib' (nosuch file), '/Applications/WezTerm.app/Contents/MacOS/libEGL.dylib' (no such file), libEGL.dylib: dlopen(libEGL.dylib, 0x0002): tried: 'libEGL.dylib' (relative path not allowed in hardened program), '/System/Volumes/Preboot/Cryptexes/OSlibEGL.dylib' (no such file), '/usr/lib/libEGL.dylib' (no such file, not in dyld cache), 'libEGL.dylib' (relative path not allowed in hardened program), '/usr/lib/libEGL.dylib' (no such file, not in dyld cache), falling back to CGL
22:43:30.108  DEBUG  wezterm_gui::termwindow       > OpenGL initialized! AMD Radeon RX 6900 XT OpenGL Engine 4.1 ATI-4.12.7 wezterm version: 20230408-112425-69ae8472
```

and ```
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform Control(LineFeed)
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform PrintString("22:43:30.069  DEBUG  wezterm_gui::termwindow       > Resized { dimensions: Dimensions { pixel_width: 984, pixel_h
eight: 780, dpi: 144 }, window_state: (empty), live_resizing: false }")
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Sgr(Reset))
22:52:48.414  DEBUG  wezterm_term::terminalstate            > Reset
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform Control(LineFeed)
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform PrintString("22:43:30.074  DEBUG  wezterm_gui::termwindow       > NeedRepaint")
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Sgr(Reset))
22:52:48.414  DEBUG  wezterm_term::terminalstate            > Reset
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform Control(CarriageReturn)
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform Control(LineFeed)
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform PrintString("22:43:30.076  DEBUG  window::os::macos::window     > EGL init failed: with_egl_lib failed: /Applications/WezTerm.app/Contents/MacOS/libEGL.dylib: dlopen(/Applications/WezTerm.app/Contents/MacOS/libEGL.dylib, 0x00")
22:52:48.414  DEBUG  wezterm_term::terminalstate::performer > perform CSI(Sgr(Reset))
```

The last second before total lock up last night was

```
00:32:11.299  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=15.088696ms
00:32:11.299  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=15.254406ms, fps=0.015340969
00:32:11.301  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.301  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.301  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.302  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.304  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.305  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.305  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.306  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.306  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.306  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.306  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.306  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.307  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:11.320  DEBUG  wezterm_gui::termwindow                > NeedRepaint
00:32:11.334  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl before call_draw elapsed=13.152112ms
00:32:11.334  DEBUG  wezterm_gui::termwindow::render::paint > paint_impl elapsed=13.405421ms, fps=0.015340969
00:32:12.284  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:12.284  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
00:32:12.284  DEBUG  wezterm_gui::termwindow                > Notification(Any { .. })
```

**Comment by wez** [comment]
If the OS panics then it is an OS bug; doesn't matter if it correlates to wezterm running or not; wezterm should not be able to cause the OS to panic.

I notice that you are using the `Software` renderer; I do not recommend it, as it is the slowest mode of operation.  Leave the front_end selection to the default unless you have good reason to change it.

**Reply by yeled** [reply]
I was trying the `Software` to see if there was a GPU bug. It seems to happen either way.

Strange that today the crashing happened only at the end of the day (like about now).
